bin/l1jetanalysis.cxx

```diff
@@ -0,0 +1,507 @@
+#include "L1Trigger/L1TNtuples/interface/L1AnalysisRecoMetDataFormat.h"
+#include "L1Trigger/L1TNtuples/interface/L1AnalysisRecoMetFilterDataFormat.h"
+
+/* TODO: put errors in rates...
+creates the rates and distributions for l1 trigger objects
+How to use:
+1. input the number of bunches in the run (~line 35)
+2. change the variables "newConditionsNtuples" and "oldConditionsNtuples" to ntuple paths
+3. If good run JSON is not applied during ntuple production, modify isGoodLumiSection()
+
+Optionally, if you want to rescale to a given instantaneous luminosity:
+1. input the instantaneous luminosity of the run (~line 32) [only if we scale to 2016 nominal]
+2. select whether you rescale to L=1.5e34 (~line606??...) generally have it setup to rescale
+nb: for 2&3 I have provided the info in runInfoForRates.txt
+*/
+
+// configurable parameters
+double numBunch = 1537; //the number of bunches colliding for the run of interest
+double runLum = 0.02; // 0.44: 275783  0.58:  276363 //luminosity of the run of interest (*10^34)
+double expectedLum = 1.15; //expected luminosity of 2016 runs (*10^34)
+
+void jetanalysis(bool newConditions, const std::string& inputFileDirectory);
+
+int main(int argc, char *argv[])
+{
+  bool newConditions = true;
+  std::string ntuplePath("");
+
+  if (argc != 3) {
+    std::cout << "Usage: l1jetanalysis.exe [new/def] [path to ntuples]\n"
+	      << "[new/def] indicates new or default (existing) conditions" << std::endl;
+    exit(1);
+  }
+  else {
+    std::string par1(argv[1]);
+    std::transform(par1.begin(), par1.end(), par1.begin(), ::tolower);
+    if(par1.compare("new") == 0) newConditions = true;
+    else if(par1.compare("def") == 0) newConditions = false;
+    else {
+      std::cout << "First parameter must be \"new\" or \"def\"" << std::endl;
+      exit(1);
+    }
+    ntuplePath = argv[2];
+  }
+
+  jetanalysis(newConditions, ntuplePath);
+
+  return 0;
+}
+
+// only need to edit this section if good run JSON
+// is not used during ntuple production
+bool isGoodLumiSection(int lumiBlock)
+{
+  if (lumiBlock >= 1
+      || lumiBlock <= 10000) {
+    return true;
+  }
+
+  return false;
+}
+
+double deltaPhi(double phi1, double phi2) {
+  double result = phi1 - phi2;
+  if(fabs(result) > 9999) return result;
+  while (result > TMath::Pi()) result -= 2*TMath::Pi();
+  while (result <= -TMath::Pi()) result += 2*TMath::Pi();
+  return result;
+}
+
+double deltaR(double eta1, double phi1, double eta2, double phi2) {
+  double deta = eta1 - eta2;
+  double dphi = deltaPhi(phi1, phi2);
+  return sqrt(deta*deta + dphi*dphi);
+}
+
+void jetanalysis(bool newConditions, const std::string& inputFileDirectory){
+  
+  bool hwOn = true;   //are we using data from hardware? (upgrade trigger had to be running!!!)
+  bool emuOn = true;  //are we using data from emulator?
+  //for efficiencies & resolutions:
+  bool recoOn = true; //are we using reco quantities? 
```

Comment:
Hi @christopheralanwest 

I have update the PR here, fixing the lines displaying the L1 Met resolution plots, and it should be fine now . Regarding the RECO information needed in the scripts: recoOn should always be True , since this is needed for both Efficiency and resolution plots. So I keep it as is and no change in the scripts is needed .
Cheers,
Georgia